raft/raft.go

```diff
@@ -565,15 +565,25 @@ func (r *raft) Step(m pb.Message) error {
 	case m.Term > r.Term:
 		lead := m.From
 		if m.Type == pb.MsgVote {
+			if r.checkQuorum && r.state != StateCandidate && r.electionElapsed < r.electionTimeout {
```

Comment:
Add a comment like: "Section 4.2.3: 'if a server receives a RequestVote request within the minimum election timeout of hearing from a current leader, it does not update its term or grant its vote.' This is tied to the checkQuorum feature because this is the only case in which leaders track the time elapsed since they last confirmed their leadership. In StateCandidate, r.electionElapsed is small, but by definition we have not recently heard from the current leader if we are in this state"